modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/PreferenceUI.java

```diff
@@ -276,11 +276,16 @@ public void setPreferencePageProviders(@Service List<PreferencePageProvider> pro
 		this.providers.setAll(providers);
 	}
 
-	private Predicate<PreferencePageProvider> getFilter() {
+	protected Predicate<PreferencePageProvider> getFilter() {
 		String filter = currentFilter == null ? "*" : "*" + currentFilter + "*"; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
 		return provider -> provider.select(filter);
 	}
 
+	protected Comparator<PreferencePageProvider> getComparator() {
```

Comment:
we could think about passing this information to a protected constructor - it is in general bad design to call subclass overrideable methods as part of the constructor.

Reply:
I updated the code, such that the filter and comparator can be passed through the constructor.